platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -187,6 +187,9 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
         registration.add(BuildFeatures::class.java, buildFeatures)
         registration.add(BuildActionModelRequirements::class.java, requirements)
         registration.addProvider(SharedBuildTreeScopedServices())
+        registration.add(BeanStateWriterLookup::class.java)
```

Comment:
❓ Why we lifted these services out of `isConfigurationCache` condition, if IP is working a top of CC?
Or that `IsolatedAction` is not a something that intended to work only in IP mode?

Reply:
I added some context to the PR description. 

To give users a consistent experience and a direct path to _Isolated Projects_ compatibility, `IsolatedAction`s are always isolated regardless of whether Isolated Projects is enabled.